### PR TITLE
Ragam page

### PR DESCRIPTION
**Pull request by shaaza**
- Added API endpoint for ragam page.
- Changed query param names for search query.
- Put deploy scripts in source.

### CODE REVIEW

File: src/kosha/app/ragam.clj [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    (when parent-ragam-id (db/ragam parent-ragam-id))))
+
+(defn handler
+  "Retrieves details of a ragam and its kritis, given the request containing ragam-id param."
+  [{:keys [params] :as request}]
+  (let [ragam-id     (Integer/parseInt (:ragam-id params))
+        ragam        (db/ragam ragam-id)
+        kritis       (db/kritis-of-ragam ragam-id)
+        parent-ragam (parent-ragam ragam)]
+    (util/json-response (response-body ragam kritis parent-ragam))))
```
Comment (ssrihari) [inline src/kosha/app/ragam.clj]:
```clojure
(let [ragam-id (Integer/parseInt (:ragam-id params))]
  (->> {:ragam        (db/ragam ragam-id)
        :kritis       (db/kritis-of-ragam ragam-id)
        :parent-ragam (parent-ragam ragam)}
       util/json-response))
```

File: src/kosha/app/ragam.clj [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  (:require [kosha.db.ragam :as db]
+            [kosha.app.util :as util]))
+
+(defn response-body
+  "Join ragam data, its kritis and its parent ragam into a single map."
+  [ragam kritis parent]
+  (-> {}
+      (assoc :ragam ragam)
+      (assoc :kritis kritis)
+      (assoc :parent-ragam parent)))
```
Comment (ssrihari) [inline src/kosha/app/ragam.clj]: I'd avoid this fn altogether (see comment below). But if I had to write it, I'd write a literal map:
```clojure
{:ragam ragam
 :kritis kritis
 :parent-ragam parent}
```

File: src/kosha/app/search.clj [hunk 3]
```diff
@@ -3,13 +3,14 @@
             [kosha.app.util :as util]))
 
 
-(defn get-ragam
+(defn get-ragams
   "Retrieves 10 best matches for given ragam name."
-  [{:keys [params] :as request}]
-  (let [ragams (db-search/ragams (:q params) 10)]
+  [query]
+  (let [ragams (db-search/ragams query 10)]
```
Comment (ssrihari) [inline src/kosha/app/search.clj]: If you already have a config, pull this `10` out into that. At the least, pull it out as a global var in this ns. See http://wiki.c2.com/?MagicNumber.

File: src/kosha/app/util.clj [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 
 (defn json-response [response]
   {:status 200
    :body (json/encode response)
    :content-type "application/json"})
 
-(defonce config (cfg/load "resources/config.edn"))
-(defn get-config [& keys]
-  (apply (partial cfg/get config) keys))
+(defonce config (config/load "resources/config.edn"))
```
Comment (ssrihari) [inline src/kosha/app/util.clj]: ah, I now see why you'd chosen `cfg`, it is shadowed here. Maybe use `config-map`, or `config-store` here?

File: src/kosha/db/ragam.clj [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    (first (j/query db-pool/conn q :identifier db-util/->hyphens))))
+
+(defn kritis-of-ragam
+  "Retrieves all kritis of a ragam with the given id."
+  [ragam-id]
+  (let [q ["SELECT kriti_id, name, composer, taala, language, ragam_id, url, data_source
+            FROM kritis
+            WHERE ragam_id = ?; "
+           ragam-id]]
+    (vec (j/query db-pool/conn q :identifer db-util/->hyphens))))
```
Comment (ssrihari) [inline src/kosha/db/ragam.clj]: Might want to consider extracting this last line out into a fn, depending how often it's repeated.

File: src/kosha/app/search.clj (hunk 3 above)
Comment (shaaza) [reply src/kosha/app/search.clj]: Do you think it's worthwhile now to add pagination to the search endpoint, and use this number as a maximum when pagination range is unspecified?

Comment (ssrihari) [reply src/kosha/app/search.clj]: No, let's not paginate server side now. If we need pagination for UX reasons, we can utilize the pagination feature in `reactable`.

**Pull request merged**